src/newgrf_gui.cpp

```diff
@@ -145,7 +145,7 @@ static void ShowNewGRFInfo(const GRFConfig *c, const Rect &r, bool show_params)
  */
 struct NewGRFParametersWindow : public Window {
 	static GRFParameterInfo dummy_parameter_info; ///< Dummy info in case a newgrf didn't provide info about some parameter.
-	GRFConfig *grf_config; ///< Set the parameters of this GRFConfig.
+	GRFConfig &grf_config; ///< Set the parameters of this GRFConfig.
```

Comment:
This makes the class non-copyable/non-moveable, which I first thought would be an issue as we would never be able to make a `std::vector<Window>`, but we can't any how due to inheritance. And making a copy of this window seems very unlikely. So I guess it's fine here.

Reply:
Hmm, keeping references like this is perhaps a bit naughty -- but on the other hand a pointer could also be invalidated anyway.

A local copy of the GRFConfig could work, but updating the real parameters would then be more fiddly.